views/assets/js/App/Collapse.js

```diff
@@ -0,0 +1,68 @@
+class Collapse {
+  constructor() {
+    this.toggleElements = Array.prototype.slice.call(
+      document.querySelectorAll('[data-toggle="collapse"]')
+    );
+    this.collapsibleElements = Array.prototype.slice.call(
+      document.querySelectorAll(".collapse")
+    );
+    this.handleClick = this.handleClick.bind(this);
+    this.init();
+  }
+  handleClick(event) {
+    const targetId =
+      event.target.dataset.target || event.target.getAttribute("href");
+    const targetElement = document.querySelector(targetId);
+    if (!targetElement) {
+      return false;
+    }
+
+    // Collapse this element
+    if (targetElement.getAttribute("aria-expanded") === "true") {
+      targetElement.setAttribute("aria-expanded", "false");
+      targetElement.style.maxHeight = 0;
+      return true;
+    }
+
+    // Expand this element
+    const naturalHeight = parseInt(targetElement.dataset.naturalHeight);
+    if (isNaN(naturalHeight) || !naturalHeight) {
+      return false;
+    }
+    targetElement.setAttribute("aria-expanded", "true");
+    targetElement.style.maxHeight = naturalHeight + "px";
+
+    return true;
+  }
+  init() {
+    var instance = this;
+
+    // Handle click events on the elements that toggle
+    this.toggleElements.forEach(function (element) {
+      element.addEventListener("click", function (event) {
+        if (instance.handleClick(event)) {
+          event.preventDefault();
+        }
+      });
+    });
+
+    // Keep track of the natural heights of the collapsed elements
+    this.collapsibleElements.forEach(function (element) {
+      const initiallyVisible =
+        Array.prototype.indexOf.call(element.classList, "show") !== -1;
```

Comment:
use `element.classList.contains("show")` instead. It will be a lot more readable.